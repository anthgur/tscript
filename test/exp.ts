//
// test precedence and associativity of operators
//

print 1+2*3;
print 3*2+1;
print (1+2)*3;
print 1+2+3;
print 2*3*4;

var abc;
var def;

print abc = def = 42;
print abc + def;

print abc = 1 + 2 * 3;

print abc = 1 - 2 + 3;
print 2 - 2 * 2;

// sanity check for literals blowing up

0;
0.0;
.0;
0.5e+10;
123E-14;

0xabcdef;
0xABCEDF;
0x01234;
0x56789;