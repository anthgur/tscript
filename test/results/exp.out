7
7
9
6
24
5
5
42
84
7
2
-2
